pkg/chains/chain.go

```diff
@@ -87,99 +76,68 @@ func (chain Chain) EncodeAddress(b []byte) (string, error) {
 	return "", fmt.Errorf("chain (%d) not supported", chain.ChainId)
 }
 
-func (chain Chain) BTCAddressFromWitnessProgram(witnessProgram []byte) (string, error) {
-	chainParams, err := GetBTCChainParams(chain.ChainId)
-	if err != nil {
-		return "", err
-	}
-	address, err := btcutil.NewAddressWitnessPubKeyHash(witnessProgram, chainParams)
-	if err != nil {
-		return "", err
-	}
-	return address.EncodeAddress(), nil
-}
-
-// DecodeAddress decode the address string to bytes
-func (chain Chain) DecodeAddress(addr string) ([]byte, error) {
-	return DecodeAddressFromChainID(chain.ChainId, addr)
-}
-
 // DecodeAddressFromChainID decode the address string to bytes
-func DecodeAddressFromChainID(chainID int64, addr string) ([]byte, error) {
-	if IsEVMChain(chainID) {
+// additionalChains is a list of additional chains to search from
+// in practice, it is used in the protocol to dynamically support new chains without doing an upgrade
+func DecodeAddressFromChainID(chainID int64, addr string, additionalChains []Chain) ([]byte, error) {
+	if IsEVMChain(chainID, additionalChains) {
 		return ethcommon.HexToAddress(addr).Bytes(), nil
-	} else if IsBitcoinChain(chainID) {
+	} else if IsBitcoinChain(chainID, additionalChains) {
 		return []byte(addr), nil
 	}
 	return nil, fmt.Errorf("chain (%d) not supported", chainID)
 }
 
 // IsEVMChain returns true if the chain is an EVM chain or uses the ethereum consensus mechanism for block finality
-func IsEVMChain(chainID int64) bool {
-	return ChainIDInChainList(chainID, ChainListByConsensus(Consensus_ethereum))
+// additionalChains is a list of additional chains to search from
+// in practice, it is used in the protocol to dynamically support new chains without doing an upgrade
+func IsEVMChain(chainID int64, additionalChains []Chain) bool {
```

Comment:
For this function and following ones it could be justified to use variadic parameter `...Chain` to simplify the signature and make it more visual when not providing chain

The motivation keeping a required argument is that, in practice, outside of tests, we'll always provide a custom value.
- GetChainList from authority in the protocol
- AdditionalChains in ZetaClient
Only in test as placeholder we provide `[]Chain{}`